File: C++/135-Candy.cpp
// Runtime: 28 ms, faster than 66.93% of C++ online submissions for Candy.
// Memory Usage: 10.6 MB, less than 30.77% of C++ online submissions for Candy.
class Solution {
public:
    int candy(vector<int>& ratings) {
        vector<int> candys(ratings.size(), 1);
        for (int i = 1; i < ratings.size(); ++i)
            if (ratings[i] > ratings[i - 1])
                candys[i] = candys[i - 1] + 1;
            else
                candys[i] = 1;
        for (int i = ratings.size() - 2; i >= 0; --i)
            if (ratings[i] > ratings[i + 1])
                candys[i] = max(candys[i], candys[i + 1] + 1);
        return accumulate(candys.begin(), candys.end(), 0);
    }
};

File: Python3.x/135-Candy.py
# Runtime: 184 ms, faster than 75.36% of Python3 online submissions for Candy.
# Memory Usage: 16.3 MB, less than 40.00% of Python3 online submissions for Candy.
class Solution:
    def candy(self, ratings: List[int]) -> int:
        candys = [1] * len(ratings)
        # left -> right
        for i in range(1, len(ratings)):
            if ratings[i] > ratings[i - 1]:
                candys[i] = candys[i - 1] + 1
        # right -> left
        for i in range(len(ratings) - 2, -1, -1):
            if ratings[i] > ratings[i + 1]:
                candys[i] = max(candys[i], candys[i + 1] + 1)
        return sum(candys)
